File: frontend/src/types/report.ts
export interface ReportRequest {
  facility_id: string;
  report_type: 'Incident Report' | 'Daily Heat Summary' | 'Facility Risk Report';
  notes?: string;
}

export interface ReportResponse {
  id: string;
  facility_id: string;
  facility_name: string;
  report_type: string;
  title: string;
  generated_at: string;
  risk_level: string;
  risk_score: number;
  summary: string;
  environmental_snapshot: Record<string, any>;
  key_findings: string[];
  actions_taken: string[];
  generated_by: string;
}

File: backend/app/schemas/report.py
from pydantic import BaseModel
from typing import List

class ReportRequest(BaseModel):
    facility_id: str
    report_type: str = "Incident Report"  # Incident Report, Daily Heat Summary, Facility Risk Report
    notes: str = ""

class ReportResponse(BaseModel):
    id: str
    facility_id: str
    facility_name: str
    report_type: str
    title: str
    generated_at: str
    risk_level: str
    risk_score: float
    summary: str
    environmental_snapshot: dict
    key_findings: List[str]
    actions_taken: List[str]
    generated_by: str = "HeatShield AI Decision Engine"
